tests/test_downloader_handlers.py

```diff
@@ -118,6 +120,35 @@ def render(self, request):
         return request.requestHeaders.getRawHeaders(b"content-length")[0]
 
 
+class ChunkedResource(resource.Resource):
```

Comment:
That's a bit tricky but I added a broken chunked resource with accompanying tests: https://github.com/scrapy/scrapy/pull/2590/files#diff-d6a30363eaf8467303835b3adce0af7dR134

It relies on a internal `request.chunked` flag which, if its purpose changes in further twisted releases, it will break the tests. I verified the behavior by using tcpdump:

* This is a normal chunked response

```
127.000.000.001.60134-127.000.000.001.60217: HTTP/1.1 200 OK
Transfer-Encoding: chunked
Content-Type: text/html
Server: TwistedWeb/16.6.0
Date: Wed, 01 Mar 2017 14:25:57 GMT

8
chunked
8
content

0

```

* This is a broken chunked response:
```
127.000.000.001.60720-127.000.000.001.60754: HTTP/1.1 200 OK
Transfer-Encoding: chunked
Content-Type: text/html
Date: Wed, 01 Mar 2017 14:30:42 GMT
Server: TwistedWeb/16.6.0

8
chunked
8
content


```